### PR TITLE
Upgrading to React 18

### PR DESCRIPTION
**Pull request by mattgperry**

### CODE REVIEW

File: packages/framer-motion-3d/src/components/use-layout-camera.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ import { clamp } from "popmotion"
 const calcBoxSize = ({ x, y }: Box) => ({
     width: calcLength(x),
     height: calcLength(y),
+    top: 0,
+    left: 0,
```
Comment (nvh) [inline packages/framer-motion-3d/src/components/use-layout-camera.ts:15]: Why were these omitted before?

File: packages/framer-motion-3d/src/render/utils/set-value.ts [hunk 2]
```diff
@@ -57,6 +57,10 @@ export function setThreeValue(
     if (setters[key]) {
         setters[key](instance, values[key])
     } else {
+        if (key === "opacity" && !instance.transparent) {
+            instance.transparent = true
+        }
```
Comment (nvh) [inline packages/framer-motion-3d/src/render/utils/set-value.ts:62]: Is this a driveby fix?

File: packages/framer-motion/src/utils/__tests__/use-cycle.test.tsx [hunk 3]
```diff
@@ -23,12 +23,6 @@ describe("useCycle", () => {
         click(container.firstChild as Element)
         click(container.firstChild as Element)
 
-        /**
-         * 1 is doubled at the start of the array because in StrictMode
-         * useEffect is double-fired on mount.
-         *
-         * When resolution is changed to 18, add another 1 to start of array
-         */
         expect(results).toEqual([1, 2, 3, 4, 1])
```
Comment (nvh) [inline packages/framer-motion/src/utils/__tests__/use-cycle.test.tsx:26]: You did not implement this comment? 🤔

File: dev/examples/ThreeBox.tsx [hunk 4]
```diff
@@ -43,11 +61,9 @@ function Box(props) {
 
 export const App = () => {
     const [isHovered, setHover] = useState(false)
+    console.log({ isHovered })
```
Comment (nvh) [inline dev/examples/ThreeBox.tsx:64]: Stray console.log

File: dev/examples/ThreeBox.tsx [hunk 5]
```diff
@@ -43,11 +61,9 @@ function Box(props) {
...
     return (
-        <MotionConfig
-            isStatic
-            transition={{ type: "spring", bounce: 0, duration: 0.7 }}
-        >
+        <MotionConfig transition={{ type: "spring", bounce: 0, duration: 0.7 }}>
```
Comment (nvh) [inline dev/examples/ThreeBox.tsx:66]: Why is the `isStatic` removed?

File: dev/examples/ThreeBox.tsx [hunk 6]
```diff
@@ -11,15 +29,15 @@ function Box(props) {
...
+    // const scale = useTransform(time, (t) => Math.sin(t) * 0.5 + 1)
 
     // Subscribe this component to the render-loop, rotate the mesh every frame
     // useFrame((state, delta) => (ref.current.rotation.x += 0.01))
     // Return the view, these are regular Threejs elements expressed in JSX
     return (
         <motion.mesh
             {...props}
-            scale={scale}
+            // scale={scale}
```
Comment (nvh) [inline dev/examples/ThreeBox.tsx:40]: Are these comments intentional?

**Review by nvh — COMMENTED** [review]
There's a ton of tiny remarks, but I think it's good overall

File: packages/framer-motion-3d/src/components/use-layout-camera.ts (hunk 1 above)
Comment (mattgperry) [reply packages/framer-motion-3d/src/components/use-layout-camera.ts:15]: The type has changed in R3F

File: packages/framer-motion-3d/src/render/utils/set-value.ts (hunk 2 above)
Comment (mattgperry) [reply packages/framer-motion-3d/src/render/utils/set-value.ts:62]: No - either the latest version of Three.js or R3F doesn't set `transparent` to `true` by default.

File: packages/framer-motion/src/utils/__tests__/use-cycle.test.tsx (hunk 3 above)
Comment (mattgperry) [reply packages/framer-motion/src/utils/__tests__/use-cycle.test.tsx:26]: I changed the test instead

File: dev/examples/ThreeBox.tsx (hunk 4 above)
Comment (mattgperry) [reply dev/examples/ThreeBox.tsx]: Thanks - fixed

File: dev/examples/ThreeBox.tsx (hunk 5 above)
Comment (mattgperry) [reply dev/examples/ThreeBox.tsx:66]: I'm not sure why it was there tbh, probably mucking about with passing context via `MotionCanvas`. But it was disabling the animations in the sandbox.

File: dev/examples/ThreeBox.tsx (hunk 6 above)
Comment (mattgperry) [reply dev/examples/ThreeBox.tsx:40]: Yeah - it doesn't make sense to mix the `useTime` powered scale with animations, I could remove but it's quite interesting to play with

Comment (nvh) [reply dev/examples/ThreeBox.tsx:40]: Maybe add a comment why it's commented out, so other people know what to do with it too

**Comment by nvh** [comment]
Github won't let me approve, but here's a ✅
